src/main/java/io/trino/plugin/db2/DB2Client.java

```diff
@@ -85,16 +113,29 @@
     private static final int MAX_LOCAL_DATE_TIME_PRECISION = 9;
     private static final String VARCHAR_FORMAT = "VARCHAR(%d)";
 
+    private static final Map<Type, WriteMapping> WRITE_MAPPINGS = ImmutableMap.<Type, WriteMapping>builder()
+            .put(BOOLEAN, WriteMapping.booleanMapping("boolean", booleanWriteFunction()))
+            .put(BIGINT, WriteMapping.longMapping("bigint", bigintWriteFunction()))
+            .put(INTEGER, WriteMapping.longMapping("integer", integerWriteFunction()))
+            .put(SMALLINT, WriteMapping.longMapping("smallint", smallintWriteFunction()))
+            .put(TINYINT, WriteMapping.longMapping("tinyint", tinyintWriteFunction()))
+            .put(DOUBLE, WriteMapping.doubleMapping("double precision", doubleWriteFunction()))
+            .put(REAL, WriteMapping.longMapping("real", realWriteFunction()))
+            .put(VARBINARY, WriteMapping.sliceMapping("varbinary", varbinaryWriteFunction()))
+            .put(DATE, WriteMapping.longMapping("date", dateWriteFunctionUsingSqlDate()))
+            .build();
+
```

Comment:
Why do we need this.. can't it be included similar to the one here https://github.com/trinodb/trino/blob/5297372129b5a4f783999cfd8ca6f3e872022cd4/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java#L411-#L434

Reply:
It doesn't really matter where you put it imo, it does the same thing. This is how I found it formatted in the old source code where we were using it before.

Reply:
So this is just for alignment with what's being done. can you point me the old source code that your are referring here..

Reply:
https://github.com/trinodb/trino/blob/338425db5019102630bcfe5e6894e2677b05ebae/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java#L897

Reply:
`It doesn't really matter where you put it imo, it does the same thing`

If it does the same thing the steps here is confusing, as why do we need a map to do separately.. I'm trying to understand the difference/benefits it provides using a map ?

Reply:
The reason it's separate is because I copied and pasted exactly what we were using before without modification because I'm not super confident in my Java skills. But if you feel strongly about refactoring this code I can change it.

Reply:
I don't find it any more confusing one way or another, personally, but I can accommodate your preference.

Reply:
would prefer all the mappings to happen within a single function, instead of it being broken down to a map and rest of the mappings in a function.

Reply:
Why? What do you find confusing about it?